ranger/ext/rifle.py

```diff
@@ -355,23 +355,44 @@ def execute(self, files,  # noqa: E501 pylint: disable=too-many-branches,too-man
 
                 cmd = prefix + [command]
                 if 't' in flags:
-                    if 'TERMCMD' not in os.environ:
-                        term = os.environ['TERM']
-                        if term.startswith('rxvt-unicode'):
-                            term = 'urxvt'
-                        elif term.startswith('rxvt-'):
-                            # Sometimes urxvt calls itself "rxvt-256color"
-                            if 'rxvt' in get_executables():
-                                term = 'rxvt'
-                            else:
-                                term = 'urxvt'
-                        if term not in get_executables():
-                            self.hook_logger("Can not determine terminal command.  "
-                                             "Please set $TERMCMD manually.")
-                            # A fallback terminal that is likely installed:
-                            term = 'xterm'
-                        os.environ['TERMCMD'] = term
-                    cmd = [os.environ['TERMCMD'], '-e'] + cmd
+                    term = os.environ.get('TERMCMD', os.environ['TERM'])
+
+                    if term not in get_executables():
+                        self.hook_logger("Can not determine terminal command.  "
+                                         "Please set $TERMCMD manually.")
+                        # A fallback terminal that is likely installed:
+                        term = 'xterm'
+
+                    # Choose correct cmdflag accordingly
+                    if term.lower() in ['xfce4-terminal', 'mate-terminal',
+                                        'terminator']:
+                        cmdflag = '-x'
+                    elif term.lower() in ['xterm', 'xterm-256color', 'urxvt',
```

Comment:
Did you check whether `xterm-256color`, `rxvt-256color` and `rxvt-unicode` are ever binary names? I'm asking because that's different from the previous behavior. I believe the command for urxvt is always named urxvt or rxvt, at least that's what the previous behavior indicates.

Reply:
Yes I overlooked this part, will use 'xterm' for ['xterm', 'xterm-256color'], and 'urxvt' for ['urxvt', 'rxvt-unicode'], and for 'rxvt' for 'rxvt-256color', if 'rxvt' found in executables, use 'rxvt', otherwise use 'urxvt'. Plan to change that in the next commit.

What do you think we should do with those not working with -e or -x or whatever (e.g. pantheon-terminal), fallback to 'xterm'?

Btw I don't remember did terminology work for you? And termite?

Reply:
Don't forget to include "Fixes #1108" in a commit message so the issue gets linked and it doesn't have to be the commit where you satisfactorily solve the problem, it's a nice indication someone's working on it so people don't quietly perform redundant work. (Would also make it easier for me to refer back to it ; )

Both terminology, termite and pantheon-terminal work with `-e "vim file.txt"` for me. tilda works with `-c "vim file.txt"`. st works with `st -e vim file.txt`.
Haven't been able to test kitty because of driver woes.